test/handler.spec.ts

```diff
@@ -194,24 +198,76 @@ describe('DataHandler unit tests', () => {
     expect(debugMessages.length).to.eq(3);
   });
 
-  it('returning null in an operator should halt data handling', () => {
-    const handler = new DataHandler(DataLayerTarget.find('digitalData.page')!);
+  [null, []].forEach((val) => {
+    it(`returning ${JSON.stringify(val)} in an operator should halt data handling`, () => {
+      const handler = new DataHandler(DataLayerTarget.find('digitalData.page')!);
+
+      const seen: any[] = [];
 
-    const seen: any = [];
+      const staticReturn = new StaticReturnOperator(val);
+      const echo = new EchoOperator(seen);
+
+      handler.push(staticReturn, echo);
+      handler.fireEvent();
+
+      expect(seen.length).to.eq(0);
+    });
+  });
+
+  it('empty object should halt data handling', () => {
+    (globalThis as any).digitalData.emptyObject = { undefinedChild: undefined };
+    const handler = new DataHandler(DataLayerTarget.find('digitalData.emptyObject')!);
+
+    const seen: any[] = [];
 
-    const nullify = new NullOperator();
     const echo = new EchoOperator(seen);
 
-    handler.push(nullify, echo);
+    handler.push(echo);
     handler.fireEvent();
 
     expect(seen.length).to.eq(0);
   });
 
+  [0, '', false, null, []].forEach((val) => {
+    it(`non-empty object with child property value ${JSON.stringify(val)} should not halt data handling`, () => {
+      (globalThis as any).digitalData.nonEmptyObject = { undefinedChild: undefined, definedChild: val };
+      const handler = new DataHandler(DataLayerTarget.find('digitalData.nonEmptyObject')!);
+
+      const seen: any[] = [];
+
+      const echo = new EchoOperator(seen);
+
+      handler.push(echo);
+      handler.fireEvent();
+
+      expect(seen.length).to.eq(1);
+      expect(seen[0].definedChild).to.eq(val);
```

Comment:
should be expecting `undefinedChild` to be dropped? Maybe good to add for clarity?

Reply:
`undefinedChild` isn't dropped here. It will have a value of `undefined`, but it still still exist in the keys on `seen[0]`. I could be explicit about this behavior by adding:

```
expect(seen[0].undefinedChild).to.eq(undefined);
```

WDYT?

Reply:
I feel like it could be good to be explicit, to document the behavior, as well as preventing us from somehow unintentionally change this behavior in the future. Open to different opinions tho if anyone feels differently

Reply:
I added a couple assertions to be explicit about expected behavior (key is retained, but value is `undefined`)